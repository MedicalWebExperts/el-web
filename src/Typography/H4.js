import React from 'react';
import PropTypes from 'prop-types';

const propTypes = {
  text: PropTypes.string.isRequired,
};

const H4 = ({ text }) => (
  <h1>{text}</h1>
);

H4.propTypes = propTypes;

export default H4;
