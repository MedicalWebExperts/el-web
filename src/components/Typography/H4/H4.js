import React from 'react';
import {
  string, bool, oneOf, shape,
} from 'prop-types';

import theme from '../../../styles';

const propTypes = {
  text: string.isRequired,
  secondary: bool,
  color: oneOf(['primary', 'secondary', 'default']),
  styles: shape({}),
};

const defaultProps = {
  color: 'default',
  secondary: false,
  styles: {},
};

const H4 = ({
  text, color, secondary, styles,
}) => (
  <h4 style={styles}>
    <style jsx>
      {`
        h4 {
          color: ${theme.colors[color]};
          font-family: ${secondary ? theme.font.secondary : theme.font.primary};
          font-size: ${theme.font.size * 1.75}px;
          line-height: ${theme.font.size / 10};
          margin-top: 0;
          margin-bottom: ${theme.spacing * 1.6}px;
        }
      `}
    </style>
    {text}
  </h4>
);

H4.propTypes = propTypes;
H4.defaultProps = defaultProps;

export default H4;
