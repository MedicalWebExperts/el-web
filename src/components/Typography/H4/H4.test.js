import React from 'react';
import renderer from 'react-test-renderer';

import H4 from './H4';

describe('Component H4 Snapshot', () => {
  it('without arguments, take the default values', () => {
    const tree = renderer.create(<H4 text="Heading 4" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('primary color of the theme', () => {
    const tree = renderer.create(<H4 text="Heading 4" color="primary" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('secondary font-family of the theme', () => {
    const tree = renderer.create(<H4 text="Heading 4" secondary />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('added optional style inline prop', () => {
    const myStyles = {
      marginTop: 30,
      lineHeight: 2,
    };
    const tree = renderer.create(<H4 text="Heading 1" styles={myStyles} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
