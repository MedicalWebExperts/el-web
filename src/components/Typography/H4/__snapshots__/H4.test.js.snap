// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component H4 Snapshot added optional style inline prop 1`] = `
<h4
  className="jsx-1352789389"
  style={
    Object {
      "lineHeight": 2,
      "marginTop": 30,
    }
  }
>
  Heading 1
</h4>
`;

exports[`Component H4 Snapshot primary color of the theme 1`] = `
<h4
  className="jsx-2701408071"
  style={Object {}}
>
  Heading 4
</h4>
`;

exports[`Component H4 Snapshot secondary font-family of the theme 1`] = `
<h4
  className="jsx-150344464"
  style={Object {}}
>
  Heading 4
</h4>
`;

exports[`Component H4 Snapshot without arguments, take the default values 1`] = `
<h4
  className="jsx-1352789389"
  style={Object {}}
>
  Heading 4
</h4>
`;
